### PR TITLE
poac 0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by ken-matsui**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/poac.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on xcode: ["12.5", :build]
+  depends_on "boost"
+  depends_on :macos
+  depends_on "openssl@3"
+
+  def install
+    mkdir "bulid" do
```
Comment (SMillerDev) [inline Formula/poac.rb:17]: Why not build?

File: Formula/poac.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      system "cmake", "..", "-G Ninja", *std_cmake_args
```
Comment (SMillerDev) [inline Formula/poac.rb:18]: Why generate ninja files?

File: Formula/poac.rb (hunk 1 above)
Comment (ken-matsui) [reply Formula/poac.rb:17]: just a typo

File: Formula/poac.rb (hunk 2 above)
Comment (ken-matsui) [reply Formula/poac.rb:18]: faster than make

File: Formula/poac.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      system "ninja", "install"
+    end
+
+    man1.install Dir["src/etc/man/man1/*.1"]
+    cp "src/etc/poac.bash", "src/etc/poac.zsh"
```
[the remark this answers is not in the document]
Comment (ken-matsui) [reply Formula/poac.rb:23]: bash & zsh completions are written in the same file; they should be duplicated to install both.

[the remark this answers is not in the document]
Comment (ken-matsui) [reply Formula/poac.rb:23]: @SMillerDev 
You mean I can do like?:

```ruby
bash_completion.install "src/etc/poac.bash" => "poac"
zsh_completion.install "src/etc/poac.bash" => "_poac"
```

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/poac.rb:23]: Yeah

[the remark this answers is not in the document]
Comment (ken-matsui) [reply Formula/poac.rb:23]: @SMillerDev 

Homebrew reports:

```console
Errno::ENOENT: No such file or directory - src/etc/poac.bash
```

`install` seems to have moved (not copied) the file.
Do you have any other workaround here?

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/poac.rb:23]: You can change it to `install_symlink`

[the remark this answers is not in the document]
Comment (ken-matsui) [reply Formula/poac.rb:23]: It worked! Thank you.

File: Formula/poac.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  url "https://github.com/poacpm/poac.git",
+    tag:      "0.3.1",
+    revision: "0afee72936c8c52ab2a477f0f14b45282bae83c1"
+  license "Apache-2.0"
+  head "https://github.com/poacpm/poac.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on xcode: ["12.5", :build]
+  depends_on "boost"
+  depends_on :macos
```
Comment (SMillerDev) [inline Formula/poac.rb:13]: Why does it depend on macOS and xcode?

Comment (ken-matsui) [reply Formula/poac.rb:13]: @SMillerDev 
This application cannot be built on Ubuntu 16.04 & macOS Catalina or earlier.
Should I remove them?

Comment (SMillerDev) [reply Formula/poac.rb:13]: Why not?

Comment (ken-matsui) [reply Formula/poac.rb:13]: @SMillerDev C++20 is required

Comment (SMillerDev) [reply Formula/poac.rb:13]: So then it can just depend on gcc on Linux and require a specific macOS version.

Comment (ken-matsui) [reply Formula/poac.rb:13]: @SMillerDev 

For Linux, I think I can use the following:

```ruby
on_linux do
  depends_on "gcc"
end
```
(Is this correct?)

But, could you please tell me how to require a specific macOS version?

Comment (SMillerDev) [reply Formula/poac.rb:13]:
```rb
depends_on macos: :catalina # C++20
```

Comment (ken-matsui) [reply Formula/poac.rb:13]: Thank you!

**Comment by ken-matsui** [comment]
@SMillerDev 
It seems that homebrew provides `g++-5`. How can I change it to an up-to-date version?

https://github.com/Homebrew/homebrew-core/runs/6650433333?check_suite_focus=true#step:7:79

**Comment by ken-matsui** [comment]
Thank you :)

**Pull request closed without merging**